### PR TITLE
GH-40055: [Java][Docs] Simplify use of Filter and Expression into Dataset Substrait

### LINKED ISSUE #40055: [Java][Dataset][Substrait][Doc]: Simplify creation of SQL Expression Filter and Projections using new Substrait Feature
### Describe the enhancement requested

Simplify creation of SQL Expression Filter and Projections into Arrow Java Dataset module using new [Substrait Feature for SQL Expressions](https://github.com/substrait-io/substrait-java/releases/tag/v0.26.0).

### Component(s)

Documentation, Java

### PR DESCRIPTION
**Pull request by davisusanibar**
### Rationale for this change

Simplify creation of SQL Expression Filter and Projections into Arrow Java Dataset module using new [Substrait Feature for SQL Expressions](https://github.com/substrait-io/substrait-java/releases/tag/v0.26.0).

### What changes are included in this PR?

Update Apache Arrow Java Dataset Substrait documentation

### Are these changes tested?

Yes

### Are there any user-facing changes?

No
* Closes: #40055

### CODE REVIEW

File: docs/source/java/substrait.rst [hunk 1]
```diff
@@ -148,297 +136,60 @@ This Java program:
...
+        ByteBuffer byteBuffer = ByteBuffer.allocateDirect(expressionToByte.length);
+        byteBuffer.put(expressionToByte);
+        return byteBuffer;
       }
     }
 
 .. code-block:: text
 
-    ADD_TEN_TO_COLUMN_N_REGIONKEY	CONCAT_COLUMNS_N_NAME_AND_N_COMMENT
+    column-1	column-2
```
Comment (vibhatha) [inline docs/source/java/substrait.rst:192]: @davisusanibar I guess this column name changes occur from the Substrait end?

File: docs/source/java/substrait.rst [hunk 2]
```diff
@@ -148,297 +136,60 @@ This Java program:
...
+             DatasetFactory datasetFactory =
+                 new FileSystemDatasetFactory(
+                     allocator, NativeMemoryPool.getDefault(), FileFormat.PARQUET, uri);
+             Dataset dataset = datasetFactory.finish();
+             Scanner scanner = dataset.newScan(options);
+             ArrowReader reader = scanner.scanBatches()) {
           while (reader.loadNextBatch()) {
-            System.out.println(
-                reader.getVectorSchemaRoot().contentToTSVString());
+            System.out.println(reader.getVectorSchemaRoot().contentToTSVString());
```
Comment (vibhatha) [inline docs/source/java/substrait.rst:168]: nit: does this need formatting. I am merely asking this comparing with the existing code.

**Review by vibhatha — APPROVED** [review]
@davisusanibar thank you for working on this. PR LGTM!

File: docs/source/java/substrait.rst (hunk 1 above)
Comment (davisusanibar) [reply docs/source/java/substrait.rst:192]: Yes, it is.

I just created this issue for enhancements "Produce projection/filter extended expressions with predefined column names" https://github.com/substrait-io/substrait-java/issues/229

File: docs/source/java/substrait.rst (hunk 2 above)
Comment (davisusanibar) [reply docs/source/java/substrait.rst:168]: I'll review how a linter will be configured/implemented for Java code.

**Review by danepitkin — APPROVED** [review]
LGTM!

**Review by lidavidm — APPROVED** [review]

**Pull request merged**